libgalois/src/analytics/louvain_clustering/louvain_clustering.cpp

```diff
@@ -173,6 +173,200 @@ struct LouvainClusteringImplementation
     return prev_mod;
   }
 
+  katana::Result<double> LouvainDeterministic(
+      katana::PropertyGraph* pfg, double lower,
+      double modularity_threshold_per_round, uint32_t& iter) {
+    katana::StatTimer TimerClusteringTotal("Timer_Clustering_Total");
+    TimerClusteringTotal.start();
```

Comment:
Nit: This timer is not stopped if the algorithm exits early (due to an error). I introduced a way to fix this a bit ago. You can write:

```
      katana::StatTimer timer(...);
      katana::TimerGuard timer_guard(timer);
```

And when `timer_guard` goes out of scope, the timer will stop.

Reply:
Can you point me to an example? I will use then. Thank you.

Reply:
https://github.com/KatanaGraph/katana-enterprise/blob/c0f853e2267b34129dd78d848e07a8035c9ea81f/libquery/src/MatchQuery.cpp#L2549-L2550